### ISSUE TITLE
pip install enable results in clang error: unknown argument: '-framework ApplicationServices'

### ISSUE THREAD

**Issue by gpanders** [enthought/enable#329]
I have installed all the prerequisites as listed in the README, but am getting the following error:

```make
clang++ -bundle -undefined dynamic_lookup -L/usr/local/opt/readline/lib -L/usr/local/opt/readline/lib -L/usr/loca
l/opt/openssl/lib -L/Users/greande/.pyenv/versions/3.6.7/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/readline/
lib -L/usr/local/opt/openssl/lib -L/Users/greande/.pyenv/versions/3.6.7/lib build/temp.macosx-10.13-x86_64-3.6/build/
src.macosx-10.13-x86_64-3.6/kiva/agg/agg_wrap.o -Lbuild/temp.macosx-10.13-x86_64-3.6 -lkiva_src -lagg24_src -lfreetyp
e2_src -o build/lib.macosx-10.13-x86_64-3.6/kiva/agg/_agg.cpython-36m-darwin.so -framework ApplicationServices -frame
work OpenGL -framework Carbon                                                                                       
    clang: error: unknown argument: '-framework ApplicationServices'
    clang: error: unknown argument: '-framework OpenGL'
    error: Command "clang++ -bundle -undefined dynamic_lookup -L/usr/local/opt/readline/lib -L/usr/local/opt/readline
/lib -L/usr/local/opt/openssl/lib -L/Users/greande/.pyenv/versions/3.6.7/lib -L/usr/local/opt/readline/lib -L/usr/loc
al/opt/readline/lib -L/usr/local/opt/openssl/lib -L/Users/greande/.pyenv/versions/3.6.7/lib build/temp.macosx-10.13-x
86_64-3.6/build/src.macosx-10.13-x86_64-3.6/kiva/agg/agg_wrap.o -Lbuild/temp.macosx-10.13-x86_64-3.6 -lkiva_src -lagg
24_src -lfreetype2_src -o build/lib.macosx-10.13-x86_64-3.6/kiva/agg/_agg.cpython-36m-darwin.so -framework Applicatio
nServices -framework OpenGL -framework Carbon" failed with exit status 1                                            
    
    ----------------------------------------
Command "/Users/greande/.pyenv/versions/3.6.7/envs/pytraining/bin/python -u -c "import setuptools, tokenize;__file__=
'/private/var/folders/_7/37rngnn96gx1cn5096y_tkjr002gs1/T/pip-install-5gukinlk/enable/setup.py';f=getattr(tokenize, '
open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install -
-record /private/var/folders/_7/37rngnn96gx1cn5096y_tkjr002gs1/T/pip-record-ye9skcvm/install-record.txt --single-vers
ion-externally-managed --compile --install-headers /Users/greande/.pyenv/versions/3.6.7/envs/pytraining/include/site/
python3.6/enable" failed with error code 1 in /private/var/folders/_7/37rngnn96gx1cn5096y_tkjr002gs1/T/pip-install-5g
ukinlk/enable/
```

Other information:
```bash
$ python --version
Python 3.6.7
$ pip --version
pip 19.0.1 from ...
```

macOS High Sierra 10.13.6

**Comment by jdpatt** [comment]
Same here:
```
python --version
Python 3.7.1
pip --version
pip 18.1 from ...
```
macOS Mojave here and I've verified that the frameworks are in my PATH.

**Comment by silverjam** [comment]
This PR should fix the issue: #335

**Comment by jwiggins** [comment]
Closed by #335
